cdap-watchdog/src/main/java/io/cdap/cdap/logging/clean/LogCleaner.java

```diff
@@ -70,5 +89,95 @@ public void run() {
     long completionTime = System.currentTimeMillis();
     LOG.info("File cleanup completed, Successful file deletes - {}. Failed file deletes - {}. Log Cleanup took {} ms",
              deleteCount, failureCount, (completionTime - startTime));
+
+    return folderCleanupCount < folderCleanupBatchSize ? DEFAULT_DELAY_IN_MILLIS : FOLDER_CLEANUP_DELAY_IN_MILLIS;
+  }
+
+  private void cleanupLogFolders() {
+    LOG.info("Starting log folder cleanup");
+    try {
+      long startTime = System.currentTimeMillis();
+      LOG.debug("Starting log folder cleanup");
+      folderCleanupCount = 0;
+
+      // Exclude folders for current date and current date - 1
+      long currentTime = System.currentTimeMillis();
```

Comment:
Huh. As soon as we are doing this, we should do a non-empty cleanup as well  (e.g. for over a month files) in case there are some leftovers because of unsynchronization between DB and FS. Don't need to do it in this PR though, could you please make a deparate 6.8 JIRA for this?

Reply:
Done. Added a TODO with the JIRA.